pkg/k8sutil/k8sutil.go

```diff
@@ -566,3 +567,32 @@ func UpdateDNSPolicy(podSpec *v1.PodSpec, dnsPolicy *monitoringv1.DNSPolicy) {
 
 	podSpec.DNSPolicy = v1.DNSPolicy(*dnsPolicy)
 }
+
+// This function is responsible for the following:
+//
+// Verify that the service exists in the resource's namespace
+// If it does not exist, fail the reconciliation.
+//
+// If the ServiceName is specified and a service with the same name exists in the same namespace as the
+// resource, ensure that the custom governing service's selector matches the
+// statefulsets.
+// If it is not selected, fail the reconciliation
+// Warning: the function will panic if the resource's ServiceName is nil..
+func EnsureCustomGoverningService(ctx context.Context, namespace string, serviceName string, svcClient clientv1.ServiceInterface, selectorLabels map[string]string) error {
```

Comment:
(nit)
```suggestion
func EnsureCustomGoverningService(ctx context.Context, namespace string, serviceName string, svcClient clientv1.ServiceInterface, labels map[string]string) error {
```

Reply:
this change would conflict with `"k8s.io/apimachinery/pkg/labels"`